### PR TITLE
Unittests for GithubHook + native asyncio syntax

### PR DESCRIPTION
**Pull request by kszucs**
Resolves:
- #26 Unittests for GithubHook + native asyncio syntax
- #27 Use native async/await keywords instead of @inlineCallbacks and yield

### CODE REVIEW

**Comment by kszucs** [comment]
bear is no game

**Comment by kszucs** [comment]
@ursabot

**Comment by kszucs** [comment]
@ursabot build

File: ursabot/hooks.py [hunk 1]
```diff
@@ -64,16 +62,16 @@ def handle_issue_comment(self, payload, event):
         elif command == 'build':
             if 'pull_request' not in issue:
                 message = 'Ursabot only listens to pull request comments!'
-                yield self._post(comments_url, {'body': message})
+                await self._post(comments_url, {'body': message})
```
Comment (kszucs) [inline ursabot/hooks.py]: @fsaintjacques the native asyncio syntax makes it easier to understand

File: ursabot/factories.py [hunk 2]
```diff
@@ -79,8 +79,12 @@ def prepend_step(self, step):
                repourl='https://github.com/ursa-labs/ursabot',
                mode='full'),
     ShellCommand(command=['ls', '-lah']),
-    ShellCommand(command=['pip', 'install', 'pytest', 'flake8']),
-    ShellCommand(command=['pip', 'install', '-e', '.']),
+    ShellCommand(command=['pip', 'install', 'pytest', 'flake8', 'mock']),
```
Comment (pitrou) [inline ursabot/factories.py]: Perhaps some requirements file is useful at this point?

**Comment by kszucs** [comment]
Mocking was painful...

File: ursabot/factories.py [hunk 3]
```diff
@@ -79,8 +79,12 @@ def prepend_step(self, step):
...
+    # --no-binary buildbot is required because buildbot doesn't bundle its
+    # tests to binary wheels, but ursabot's test suite depends on buildbot's
+    # so install it from source
+    ShellCommand(command=['pip', 'install', '--no-binary', 'buildbot',
+                          '-e', '.']),
```
Comment (pitrou) [inline ursabot/factories.py]: Does the `-e` flag only apply to `.` or also to `buildbot`? For clarity, perhaps issue the `pip install -e .` command separately.

**Comment by kszucs** [comment]
@ursabot build

File: .travis.yml [hunk 4]
```diff
@@ -4,7 +4,10 @@ services:
 python:
   - 3.6
 script:
-  - pip install "pytest>=3.9" flake8 -e .
+  # --no-binary buildbot is required because buildbot doesn't bundle its tests
+  # to binary wheels, but ursabot's test suite depends on buildbot's so install
+  # it from source
+  - pip install --no-binary buildbot "pytest>=3.9" mock flake8 -e .
```
Comment (pitrou) [inline .travis.yml]: See other comments below about pip installs.

File: setup.py [hunk 5]
```diff
@@ -15,7 +20,7 @@
     setup_requires=['setuptools_scm'],
     install_requires=['click', 'dask', 'docker', 'docker-map', 'toolz',
                       'buildbot', 'treq'],
-    tests_require=['pytest>=3.9'],
+    tests_require=['pytest>=3.9', 'mock'],
```
Comment (pitrou) [inline setup.py]: Oh, I forgot: `mock` is available as [unittest.mock](https://docs.python.org/3/library/unittest.mock.html) in Python 3, you shouldn't need to install it separately.

File: ursabot/factories.py (hunk 2 above)
Comment (kszucs) [reply ursabot/factories.py]: Created issue: https://github.com/ursa-labs/ursabot/issues/28

File: ursabot/factories.py (hunk 3 above)
Comment (kszucs) [reply ursabot/factories.py]: It doesn't need to be `editable` inside the build actually.

Created issue: https://github.com/ursa-labs/ursabot/issues/29

File: setup.py (hunk 5 above)
Comment (kszucs) [reply setup.py]: It was complaining somewhy, probably because `twisted.trial` requires it.

Comment (kszucs) [reply setup.py]: Or buildbot itself: https://github.com/buildbot/buildbot/blob/master/master/setup.py#L487

Comment (kszucs) [reply setup.py]: Created another issue: https://github.com/ursa-labs/ursabot/issues/30

**Comment by kszucs** [comment]
Merging, thanks @pitrou!

**Pull request merged**